src/common/components/Recaptcha/Recaptcha.test.tsx

```diff
@@ -238,7 +238,51 @@ describe('Recaptcha component', () => {
     await waitFor(() => {
       expect(onSuccess).toHaveBeenCalledTimes(1)
       expect(onFailure).not.toHaveBeenCalled()
-      expect($log.warn).toHaveBeenCalledWith('Data was missing!')
+      expect($log.warn).toHaveBeenCalledWith('Recaptcha returned an unusual response:', {})
+    })
+  })
+
+  it('should not block gifts if action is undefined', async () => {
+    //@ts-ignore
+    global.fetch = jest.fn(() => {
+      return Promise.resolve({
+        json: () => Promise.resolve({ success: true, score: 0.9 })
+      })
+    })
+
+    onSuccess.mockImplementation(() => console.log('success after weird'))
+
+    const { getByRole } = render(
+      buildRecaptcha()
+    )
+
+    await userEvent.click(getByRole('button'))
+    await waitFor(() => {
+      expect(onSuccess).toHaveBeenCalledTimes(1)
```

Comment:
You could write, but I'm not too fussy about it as this is urgent and the other tests also have this one.
```suggestion
      expect(onSuccess).toHaveBeenCalled()
```

Reply:
I like having it as 1 time to ensure the return statements are in effect and not double submitting the gift.

Reply:
That's fine my me